pkg/steps/source.go

```diff
@@ -369,46 +369,57 @@ func isBuildPhaseTerminated(phase buildapi.BuildPhase) bool {
 }
 
 func handleBuild(ctx context.Context, buildClient BuildClient, build *buildapi.Build) error {
-	if err := buildClient.Create(ctx, build); err != nil {
-		if !kerrors.IsAlreadyExists(err) {
-			return fmt.Errorf("could not create build %s: %w", build.Name, err)
+	var buildErr error
+	attempts := 3
+	if boErr := wait.ExponentialBackoff(wait.Backoff{Duration: 1 * time.Minute, Factor: 2, Steps: attempts}, func() (bool, error) {
```

Comment:
Seems like it takes about 5 minutes for a new node to come online, this backoff gives us a chance at one new node arriving in time. Given the low cost of the kubelet rejecting us, I'd consider attempts=5 and factor=1.5 which would give us a few changes for new nodes joining the back.

Reply:
Sounds good!

Reply:
I've always found the `apimachinery` parameters confusing, but note that these values will give us:

```lua
n, d, f, t = 5, 1, 1.5, 0
print("i", "t", "d")
for i = 0, n - 1 do
    print(i, t, d)
    t, d = t + d, d * f
end
```
```
i	t	d
0	0	1
1	1	1.5
2	2.5	2.25
3	4.75	3.375
4	8.125	5.0625
```

Which will give us a single retry (the last) after a new node arrives at `t ~ 5`, unless the combined retries take more than 1.875s (I don't know how long they take and could not find an example).

Reply:
I think that we do not necessarily need to retry until there is a new node - that's scheduler's job. We just need to give the scheduler some chance to retry the bad placement that results in `OutOfCpu`. It is fine if the retried Pod ends up unschedulable, that makes it wait for the node.